### ISSUE TITLE
Port forwarding: remote server closes connection, but sshj does not close connection to client

### ISSUE THREAD

**Issue by tsoiland** [hierynomus/sshj#686]
TLDR; I have a failing unit test that I think points at a legitimate bug: [https://github.com/tsoiland/sshj/commit/4a7afbcc3ca14fc091a90f9faf74ce146c307328](https://github.com/tsoiland/sshj/commit/4a7afbcc3ca14fc091a90f9faf74ce146c307328)

I originally had a problem running docker exec (http api) through sshj. From what I've been able to gather it streams data back in the http response and probably doesn't know the content length when it writes the response headers, so the client has to wait until the connection is closed by the server. 

When I try to tunnel this through sshj the connection between my http client and sshj is never closed and therefore my http client waits forever.

This works with a direct connection to docker's http server.

It also works tunnelled through openssh-client and I can see the FIN, ACKs on both sides of the tunnel with wireshark.

**Comment by vladimirlagunov** [comment]
Hi! It can be related to #143.

**Comment by tsoiland** [comment]
1. I realized while doing some more debugging that my problem is with local port forwarding, not remote like in my linked pull request. New unit test with fix in StreamCopier.java: https://github.com/tsoiland/sshj/commit/54b2310b12d26c1638be50bb69f8b533dfea35c3

2. I looked at issue #143, but there the problem seems to be the other way around. Also uncommenting the suggested line doesn't help: `trans.write(new SSHPacket(Message.CHANNEL_EOF).putUInt32(chan.getRecipient()));`
